department-of-veterans-affairs/abd-vro
# MCP-2085 Mock BIP service

## What was the problem?

Integration with the actual BIP API is taking a long time to complete and will probably be impossible to test in local and test environments. In order to complete the remaining routes, we are creating a Mock service mimicking some expected responses from BIP. 

NOT in scope:   [Upload PDF](https://amida.atlassian.net/browse/MCP-2090)

Associated tickets or Slack threads:

- [MCP-2085](https://amida.atlassian.net/browse/MCP-2085)

## How does this fix it?[^1]
- Create a Mock service that mimics BIP responses for some specific collection IDs (see testing step for details)
- Completed the "Complete Processing" route as described in https://github.com/department-of-veterans-affairs/abd-vro/wiki/VRO-v2-Roadmap


## How to test this PR
- Collection ID = 350 with defaults: Processing Goes through until "order exam" step. Then fails because of MAS
- CollectionID = 350 with disabilityActionType": "DECREASE": Return "Not in scope"
- CollectionID = 351: Return "Missing anchor" (wrong station)
- CollectionID = 352: Return "Missing anchor" (missing special issue)
- CollectionID = 353: Does not complete processing after number of attempts is exhausted


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
